cmd/mistryd/server.go

```diff
@@ -166,26 +166,40 @@ func (s *Server) HandleIndex(w http.ResponseWriter, r *http.Request) {
 		}
 
 		for _, j := range pendingJobs {
+			bi := types.BuildInfo{}
+			biBlob, err := ioutil.ReadFile(filepath.Join(pendingPath, j.Name(), BuildInfoFname))
+			if err != nil {
+				s.Log.Printf("cannot read build_info file of job %s; %s", j.Name(), err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			err = json.Unmarshal(biBlob, &bi)
 			jobs = append(jobs, Job{
 				ID:        j.Name(),
 				Project:   p.Name(),
-				StartedAt: j.ModTime(),
-				Output:    filepath.Join(pendingPath, j.Name(), BuildLogFname),
+				StartedAt: bi.StartedAt,
 				State:     "pending"})
 		}
 
 		for _, j := range readyJobs {
```

Comment:
The body of this loop is the exact same with the one some lines above. Can we somehow get rid of this repetition?

Reply:
Introduced a `GetJobs` function and extracted the repetitive code to a buildInfo function.